File: python/leetcode/232.py
class Queue(object):
    def __init__(self):
        self.stk = []

    def push(self, x):
        if not self.stk: self.stk.append(x)
        else:
            data = self.stk.pop()
            self.push(x)
            self.stk.append(data)

    def pop(self):
        self.stk.pop()

    def peek(self):
        return self.stk[-1]

    def empty(self):
        return not self.stk

File: cpp/leetcode/232.cpp
class Queue {
public:
    void push(int x) {
        if(stk.empty()) {stk.push(x); return;}
        int num = stk.top(); stk.pop();
        push(x);
        stk.push(num);
    }

    void pop(void) {
        stk.pop();
    }

    int peek(void) {
        return stk.top();
    }

    bool empty(void) {
        return stk.empty();
    }
private:
    stack<int> stk;
};